skietbaan_fe/src/components/AddMembersGroup.js

```diff
@@ -2,41 +2,49 @@ import React, { Component } from "react";
 import { connect } from "react-redux";
 import "./groups.css";
 import { BASE_URL } from "../actions/types";
-import history from "./history";
+import { withRouter } from "react-router-dom";
+import marked from "./GroupImages/marked.png";
+import unmarked from "./GroupImages/unmarked.png";
+import back from "./GroupImages/back.png";
+
 class AddMembersGroup extends Component {
   constructor(props) {
     super(props);
     this.state = {
       posts: [],
       newArray: [],
       filterText: "",
-      selected: ""
+      selected: "",
+      count: 0
     };
     this.toggleHighlight = this.toggleHighlight.bind(this);
-    this.handleOnClick = this.handleOnClick.bind(this);
     this.onBack = this.onBack.bind(this);
     this.onChange = this.onChange.bind(this);
   }
- UNSAFE_componentWillMount() {
-    fetch( BASE_URL+"/api/Groups/list?id=" + this.props.id)
-      .then(res => res.json())
-      .then(data => {
-        this.setState({
-          posts: data.map(users => {
-            users.highlighted = false;
-            return {
-              ...users,
-              highlighted: false
-            };
-          })
+  UNSAFE_componentWillMount() {
+    if (this.props.id != 0) {
+      fetch(BASE_URL + "/api/Groups/list?id=" + this.props.id)
+        .then(res => res.json())
+        .then(data => {
+          this.setState({
+            posts: data.map(users => {
+              return {
+                ...users,
+                highlighted: false,
+                backgrnd: "white"
+              };
+            })
+          });
         });
-      });
+    } else {
+      this.props.history.push("/ViewGroups");
```

Comment:
camelCase

Reply:
UNSAFE_componentWillMount is a lifecycle method defined in React.Component :)